homeassistant/components/hue/hue_sensor.py

```diff
@@ -0,0 +1,237 @@
+
+    await async_update_bridge(None)
+
+
+async def async_update_items(hass, bridge, async_add_entities, current,
+                             binary=False):
+    """Update sensors from the bridge."""
+    import aiohue
+    from homeassistant.components.hue.binary_sensor import (
+        HueNotDarkness, HuePresence)
+    from homeassistant.components.hue.sensor import (
+        HueLightLevel, HueTemperature)
+
+    api = bridge.api.sensors
+
+    try:
+        start = monotonic()
+        with async_timeout.timeout(4):
+            await api.update()
+    except (asyncio.TimeoutError, aiohue.AiohueException) as err:
+        _LOGGER.debug('Failed to fetch sensor: %s', err)
+
+        if not bridge.available:
+            return
+
+        _LOGGER.error('Unable to reach bridge %s (%s)', bridge.host, err)
+        bridge.available = False
+
+        return
+
+    finally:
+        _LOGGER.debug('Finished sensor request in %.3f seconds',
+                      monotonic() - start)
+
+    if not bridge.available:
+        _LOGGER.info('Reconnected to bridge %s', bridge.host)
+        bridge.available = True
+
+    new_sensors = []
+    sensor_device_names = {}
+
+    # Physical Hue motion sensors present as three sensors in the API: a
+    # presence sensor, a temperature sensor, and a light level sensor. Of
+    # these, only the presence sensor is assigned the user-friendly name that
+    # the user has given to the device. Each of these sensors is linked by a
+    # common device_id, which is the first twenty-three characters of the
+    # unique id (then followed by a hyphen and an ID specific to the individual
+    # sensor).
+    #
+    # To set up neat values, and assign the sensor entities to the same device,
+    # we first, iterate over all the sensors and find the Hue presence sensors,
+    # then iterate over all the remaining sensors - finding the remaining ones
+    # that may or may not be related to the presence sensors.
+    for item_id in api:
+        if item_id in current:
+            continue
+
+        name = PRESENCE_NAME_FORMAT.format(api[item_id].name)
+        if api[item_id].type == aiohue.sensors.TYPE_ZLL_PRESENCE:
+            sensor = HuePresence(api[item_id], name, bridge)
+            sensor_device_names[sensor.device_id] = api[item_id]
+            current[item_id] = sensor
+            if binary:
+                new_sensors.append(sensor)
+
+    # Iterate again now we have all the presence sensors, and add the related
+    # sensors with nice names.
+    for item_id in api:
+        if item_id in current:
+            continue
+
+        # Work out the shared device ID, as described above
+        device_id = api[item_id].uniqueid
+        if device_id and len(device_id) > 23:
+            device_id = device_id[:23]
+        name = api[item_id].name
+        primary_sensor = None
+        if api[item_id].type == aiohue.sensors.TYPE_ZLL_LIGHTLEVEL:
+            darkness_name = name + ' is not dark'
```

Comment:
Shouldn't this be just "light level" ?

Reply:
Also please use string formatting: `"{} light level".format(name)`

Reply:
This name is for the "is dark" binary sensor (whether Hue considers the room "dark", depending on the value of the light level and the thresholds configured by the user). Happy to drop this extra sensor if you think that'd be best.

Reply:
I'm going to bin it until someone has an explicit need for this. Not MVP for sensor support.